### ISSUE TITLE
Wounds and Fatigue status icons

### ISSUE THREAD

**Issue by javierriveracastro** [spacemandev-git/fvtt-swade-toolkit#9]
Swade Tools, and betterrolls inherited it, shows a nice status icon with the wounds and the fatigue:

![example](https://i.imgur.com/4XZRS9P.png)

It is synchronized on actor update, so removing the status does nothing. It is very handy and I'd like to keep it, but it also doesn't make sense to maintain the status code in betterrolls if you are doing it.

Will you accept a PR to implement it?

**Comment by spacemandev-git** [comment]
Yes, but with the PR please keep two things in mind:
1) To PR to the develop branch, not master
2) To keep it in style of the structure of the codebase.  For example, to make this change, you likely want to extend 
https://github.com/spacemandev-git/fvtt-swade-toolkit/blob/master/src/scripts/status_effects/status_effects.ts

- Make sure to create a new setting that it's gated behind (everything should be optional in toolkit)
- make sure to add translations for the setting info in en.json
- create a new function within the status effects class, and then call that function in the constructor to start your hooks (see line 11)

**Comment by javierriveracastro** [comment]
I'll do as you wish, but wouldn't it be clearer to just add the code your hooks (walled by an if game.settings.get...)?

Otherwise I'll be duplicating lots of code (get the token, get the status). That means that "damage and fatigue as states" will be need "link status effects" enabled to work, but to me that makes sense.

The other option is that I refactor that code and take the common parts to their own methods, but I prefer to get used to your coding style first.

Or I can just duplicate code, but something inside me will cry.

**Comment by spacemandev-git** [comment]
Can you link me to the relevant code in better rolls so I can see how it works to see if it should go in the existing function or in a new function?

**Comment by javierriveracastro** [comment]
Yes I could, but I'm going to totally refactor it. Better rolls code is just a copy and paste from swade tools, and it comes with too many inherited, not necessary things.

I'll try to make a first approximation this afternoon and then we can talk with some real code in front.

**Comment by javierriveracastro** [comment]
Here is the rough code.

https://github.com/javierriveracastro/fvtt-swade-toolkit/blob/develop_wound_states/src/scripts/status_effects/status_effects.ts

There is a new method "set_level_effects". This simple updates the effect icons to the current wound and fatigue status of the actor (well, right now it only works with fatigue, but you get it).

An in the "updateToken" hook I add a line to call it:

`        if (game.settings.get("swade-toolkit", "wound-status-effects")) {
            if (data?.actorData?.data?.fatigue) {
                this.set_level_effects('token', tokenDiff._id);
            }
        }`

I did two ifs for clarity. The first checks that the settings is on, the second only calls the above method when fatigue is changed to avoid an infinite loop.

I just need to add it to updateActor and add wound management.

Does it meet your code style? Do you feel it is a good aproach?

**Comment by javierriveracastro** [comment]
I've completed it. You have a PR.

**Comment by spacemandev-git** [comment]
I'm pushing out the PR now, but two potential enhancements to this:
1)If you could make on this is to also hook onto Token create, so when a Wildcard (not NPC) token is pulled to screen, it brings with it it's existing wound and fatigue values
2) Append the Wound/Fatigue numbers into the Config.status icons list and then update the sheet if they are applied (this might be more work and I'll be happy to tackle it when I have time)

**Comment by javierriveracastro** [comment]
Second one is a great idea. A fast way to apply wounds...

**Comment by javierriveracastro** [comment]
First one done. Have to make the settings refresh the screen on change for it to work.

I will try the second one before making a PR.
